dot/telemetry/telemetry.go

```diff
@@ -46,6 +46,13 @@ type Handler struct {
 	sendMessageTimeout time.Duration
 }
 
+// Instance functions that telemetry handler instance needs to implement
+type Instance interface {
+	AddConnections(conns []*genesis.TelemetryEndpoint)
+	SendMessage(msg *Message) error
+	startListening()
```

Comment:
why the private method?

Reply:
I'm trying to control who can start the listener, this is called when the instance is created.